### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by dy** [chrisguttandin/subscribable-things#75]
Hey @chrisguttandin thanks for your response. Following the offer, I'll just share some thoughts I had in mind regarding a very similar lib. There's very similar lib btw [rx-use](https://github.com/streamich/rx-use) - would be nice mentioning it as alternative, we need to support fellows.

Basically I imagined the following UI sources:
- element attribute/attributes
- caret, pointer
- hotkey/keys
- scroll
- pinch, touch
- element classes
- element content
- cookie
- element focus state
- input values
- object property/properties accessor/observer
- location/history/hash
- local/session storage (can be observed from another frame)
- dark/light mode

That's more UI-focused and I'm not sure if that's good practice in general, but for some cases may be helpful.

Also - I had a bit of hard time using _subscribable-things_ as direct ESM dependency − I ended up porting [only a piece](https://github.com/spectjs/hyperf/blob/main/test/libs/sub-things.js). Would it be useful to have a single dist ESM module, to allow easy manual include without tooling? Like [this](https://npmfs.com/package/subscribable-things/2.0.8/build/es5/bundle.js), but minified and without CJS legacy, to simply drop-in to any project in oldschool way but as ESM.

**Comment by chrisguttandin** [comment]
Thanks a lot @dy,

I created a separate issue for mentioning the alternatives.

Is there a way to observe element attribute changes or changes of the class list or even the content?

Maybe we can use this thread for discussing possible new functions and then create a new issue for each specific function which we want to add.

I know that I should modernize the way I publish packages. Do you have of a good example which I could use as a reference?

**Comment by dy** [comment]
Yeah, it's fairly straightforward to observe attribs via [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver):

```js
(new MutationObserver(mutations => {...})).observe(targetNode,  { attributes: true, childList: false, subtree: false});
```
It would only need specific handler of `mutations`, and for content that's `childList: true`.

As for publishing packages - I personally decided to follow radically minimal style, without tooling or redundant dependencies as much as possible. Also dropped backwards-compat with CJS - lifetime isn't worth it - maintaining transient bundlers.

Examples I like: [lukeed](https://github.com/lukeed/clsx), [developit](https://github.com/developit/htm). You can check [templize](https://github.com/spectjs/templize) as simple style I follow.

**Comment by chrisguttandin** [comment]
You mean adding a new function which wraps `mutations()` with some preconfigured options?

**Comment by dy** [comment]
I imagined something like
```js
let attrValue = from(attr(element, 'attr-to-observe'))
```
That internally assigns MutationObserver to element.

**Comment by chrisguttandin** [comment]
I created #77 to add the `attribute()` function.

@dy Could you please take a look if you have a moment?

**Comment by chrisguttandin** [comment]
Now that the first item on the list is done, I wonder how we could implement observing the pointer? Would that be a global listener that emits the `target` of each event?

Or do you refer to the pointer within an input field?

**Comment by dy** [comment]
Some API examples are [touch-pinch](https://www.npmjs.com/package/touch-pinch), [touch-position](https://www.npmjs.com/package/touch-position).
So it'd look like
```
import { pointer } from 'subscribable-things'

const subscribe = pointer(target=document)
```
